plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1087,29 +1087,133 @@ public void testCaseInsensitivePredicate()
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - 0 = nationkey"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        // additive inverse
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + (-nationkey) = 0"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - nationkey = 0"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
 
-        // some databases calculate remainder instead of modulus when one of the values is negative
-        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % -nationkey = 2"))
-                .isFullyPushedDown()
-                .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
+        // constant
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + 1 = 7"))
+                .matches("VALUES (BIGINT '6', CAST('FRANCE' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - 1 = 7"))
+                .matches("VALUES (BIGINT '8', CAST('INDIA' AS varchar(25)), BIGINT '2')")
+                .isFullyPushedDown();
+
+        // column reference
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + regionkey = 26"))
+                .matches("VALUES (BIGINT '23', CAST('UNITED KINGDOM' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - regionkey = 23"))
+                .matches("VALUES (BIGINT '24', CAST('UNITED STATES' AS varchar(25)), BIGINT '1')")
+                .isFullyPushedDown();
+        // column reference with AND
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE regionkey = 0 AND nationkey + regionkey = 14"))
+                .matches("VALUES (BIGINT '14', CAST('KENYA' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE regionkey = 0 AND nationkey - regionkey = 16"))
+                .matches("VALUES (BIGINT '16', CAST('MOZAMBIQUE' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
 
-        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2", errorMessageForDivisionByZeroPushdown());
```

Comment:
removed?

Reply:
simplified to `SELECT nationkey, name, regionkey FROM nation WHERE nationkey % 0 = 0`. i.e. it doesn't matter what's on the LHS here. Subtraction now has it's own test cases so no need to combine them.

Reply:
Do we still test combinations of arithmetics (like here: subtraction within modulo; or something else)?